### ISSUE TITLE
UI: nicer flag display

### ISSUE THREAD

**Issue by martinbonnin** [HexaMonkey/hexamonkey#3]
It would be nice to know the meaning of individual flags

**Comment by sevsouv** [comment]
Hi Martin,

You can split a flag into smaller integers (uint(1), uint(2)...) and use the new enum type (available in next version) to add notes to explain the value of each flag. For instance : 

	enum(uint(2, 2)) {
		0b00 : "No more than 128 frames and 1MB total tag size",
		0b01 : "No more than 64 frames and 128 KB total tag size",
		0b10 : "No more than 32 frames and 40 KB total tag size",
		0b11 : "No more than 32 frames and 4 KB total tag size"
	} tagSizeRestrictions;
	
	enum(uint(1, 2)) {
		0b0 : "No restrictions",
		0b1 : "Only ISO-8859-1 or UTF-8"
	} textEncodingRestrictions;
	
	enum(uint(2, 2)) {
		0b00 : "No restrictions",
		0b01 : "No string longer than 1024 chars",
		0b10 : "No string longer than 128 chars",
		0b11 : "No string longer than 30 chars",
	} textFieldSizeRestrctions;
	
	enum(uint(1, 2)) {
		0b0 : "No restrictions",
		0b1 : "Image are encoded only with PNG or JPEG"
	} imageEncodingRestrictions;
	
	enum(uint(2, 2)) {
		0b00 : "No restrictions",
		0b01 : "All images are 256x256 pixels or less",
		0b10 : "All images are 64x64 pixels or less",
		0b11 : "All images are exactily 64x64 pixels"
	}) imageSizeRestrictions;

Is that agreeable ?

**Comment by martinbonnin** [comment]
That works as far as the hmdl is concerned. My remark was more about the UI though. Can we have something like wireshark ?

![](http://www.lovemytool.com/.a/6a00e008d9577088340133f2181333970b-pi)

**Comment by sevsouv** [comment]
I agree that this would be nice to have. I'll try to see how to integrate it sensibly.

**Comment by sevsouv** [comment]
Added bitset view for classes, declare as

    class MP3Header with {displayMode : "bitset"}

![hexamonkey-bitset](https://cloud.githubusercontent.com/assets/3912984/9316749/3f8a3c1e-4537-11e5-978a-a271edad6ac1.png)

32 bits might be a bit much, but still readable I think.

**Comment by martinbonnin** [comment]
:+1:  :)
